source/common/common/matchers.cc

```diff
@@ -161,5 +161,23 @@ bool PathMatcher::match(const absl::string_view path) const {
   return matcher_.match(Http::PathUtil::removeQueryAndFragment(path));
 }
 
+const std::string& PathMatcher::criterion() const {
```

Comment:
If possible I think the pattern where all matcher pattern types are explicitly listed here should be avoided, as it introduces another place in the code where one needs to add a case when adding a new matcher type (if any will be added in the future).

Can this instead be done either by directly fetching the correct matcher field in the places where `criterion()` is called? (for example: in `PrefixRouteEntryImpl` the call to `matcher()` will directly return `path_matcher_->matcher_.matcher().prefix()`, [maybe matcher_ will need a const getter])

Reply:
Yeah, that sounds better, I agree. Although the accessors become the long chain of "matchers" :) `path_matcher_->matcher().matcher().prefix()`. Will change.